activesupport/lib/active_support/descendants_tracker.rb

```diff
@@ -77,15 +77,17 @@ def initialize_copy(orig)
       end
 
       def <<(klass)
-        cleanup!
         @refs << WeakRef.new(klass)
       end
 
       def each
-        @refs.each do |ref|
+        @refs.reject! do |ref|
           yield ref.__getobj__
+          false
         rescue WeakRef::RefError
+          true
         end
+        self
```

Comment:
There is a minor change here, previously `@refs` would be returned, but I believe returning `self` is much more correct.